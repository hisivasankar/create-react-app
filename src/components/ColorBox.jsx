import React, {Component} from 'react';

export function getSizeText(x, y) {
    return `${x}x${y}`;
  }
  
export default class ColorBox extends Component {
    render() {
        let style = {
            backgroundColor: this.props.color,
            width: `${this.props.width}px`,
            height: `${this.props.height}px`
        }, sText = getSizeText(this.props.width, this.props.height);
        return (<div style={style}>{sText}</div>);
    }
}
  