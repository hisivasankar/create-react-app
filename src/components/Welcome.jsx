import React, {Component} from 'react';

import {sayHello} from '../api/Utils';

export class Welcome extends Component {
    render() {
        let sMessage = sayHello({ name: this.props.name});
        return (
            <div className="welcome">{sMessage}</div>
        );
    }
}