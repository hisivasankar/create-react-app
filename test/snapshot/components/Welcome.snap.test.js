import React from 'react';
import renderer from 'react-test-renderer';
import {Welcome} from '../../../src/components/Welcome';

describe('Welcome Component Snapshot test', () => {
    it('should render 1 <div> with given name property', () => {
        const oWelcomeComp = renderer.create(
            <Welcome name='Siva'/>
        );

        let tree = oWelcomeComp.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

