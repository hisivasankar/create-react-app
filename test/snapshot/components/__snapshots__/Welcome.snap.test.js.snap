// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component Snapshot test should render 1 <div> with given name property 1`] = `
<div
  className="welcome"
>
  Hello Siva! Welcome to my website
</div>
`;
